source/miniFastLED.h

```diff
@@ -34,6 +34,45 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 #include "board.h"
 #include "hal.h"
 #include "light_utils.h"
+#include "settings.h"
+
+/*
+    HSV Constants
+*/
+#define HUE_RED 0
+#define HUE_YELLOW 32
+#define HUE_GREEN 64
+#define HUE_CYAN 96
+#define HUE_BLUE 128
+#define HUE_MAGENTA 160
+
+/*
+    Helpers for dimming. There're used by hsv2rgb internally, but you can use
+    them if you're using direct RGB colors and want to dim them naively.
+ */
+
+/* Dim a single value: R, G, B or a saturation component of the HSV color */
+static inline int8_t dimValue(uint8_t value) {
+  const uint8_t dimBy = ledIntensity * 30;
+  if (dimBy > value)
+    return 0;
+  return value - dimBy;
+}
+
+/* Naively dim all values within the led_t variable */
+static inline void naiveDimLed(led_t *color) {
+  color->p.blue = dimValue(color->p.blue);
+  color->p.red = dimValue(color->p.red);
+  color->p.green = dimValue(color->p.green);
+}
+
+/* Returns a naively "dimmed" RGB color */
+static inline uint32_t naiveDimRGB(uint32_t color) {
```

Comment:
Unsure about naming (naive/hsv dimming), and unsure if it's worth keeping in header with static inline prefix.

Reply:
For sure keep the `static inline` attribute. The CI do produce and archive a disassembly. 

for example this is a excerpt from the "green" function,
```C
  setAllKeysColor(currentKeyLedColors, naiveDimRGB(0x00FF00));
    4e72:	f000 fd8d 	bl	5990 <setAllKeysColor>
```
as you can see the dimming function is actually entirely precomputed since the input argument is constant. And all function that has implementation inside a header file should be static.

Reply:
Cool.